salt/_modules/metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,128 @@
+
+__virtualname__ = "metalk8s_kubernetes_cronjob"
+
+
+def list_cronjobs(
+    suspended=None,
+    all_namespaces=False,
+    namespace="default",
+    **kwargs,
+):
+    results = __salt__["metalk8s_kubernetes.list_objects"](
+        kind="CronJob",
+        apiVersion="batch/v1",
+        all_namespaces=all_namespaces,
+        namespace=namespace,
+        **kwargs,
+    )
+    if not results:
+        raise CommandExecutionError(
+            "No cronjobs found using the given criteria: "
+            f"suspended={suspended}, "
+            f"all_namespaces={all_namespaces}, "
+            f"namespace={namespace}"
+        )
+
+    if suspended is not None:
+        if suspended:
+            results = [cronjob for cronjob in results if cronjob["spec"]["suspend"]]
+        else:
+            results = [cronjob for cronjob in results if not cronjob["spec"]["suspend"]]
+
+    return results
+
+
+def get_cronjob(name, namespace, **kwargs):
+    cronjob = __salt__["metalk8s_kubernetes.get_object"](
+        kind="CronJob",
+        apiVersion="batch/v1",
+        name=name,
+        namespace=namespace,
+        **kwargs,
+    )
+    if not cronjob:
+        raise CommandExecutionError(
+            f"Cronjob {name} not found in namespace {namespace}"
+        )
+
+    return cronjob
+
+
+def _set_cronjob_suspend(name, namespace, suspend, **kwargs):
+    cronjob = get_cronjob(name, namespace, **kwargs)
+
+    if cronjob["spec"]["suspend"] != suspend:
+        __salt__["metalk8s_kubernetes.update_object"](
+            kind="CronJob",
+            apiVersion="batch/v1",
+            name=name,
+            namespace=namespace,
+            patch={"spec": {"suspend": suspend}},
+            old_object=cronjob,
+            **kwargs,
+        )
+
+    return True
+
+
+def suspend_cronjob(name, namespace, **kwargs):
+    return _set_cronjob_suspend(name, namespace, suspend=True, **kwargs)
+
+
+def activate_cronjob(name, namespace, **kwargs):
+    return _set_cronjob_suspend(name, namespace, suspend=False, **kwargs)
+
+
+def get_jobs(cronjob_name, namespace, **kwargs):
+    cronjob = get_cronjob(cronjob_name, namespace, **kwargs)
+
+    jobs = __salt__["metalk8s_kubernetes.list_objects"](
+        kind="Job", apiVersion="batch/v1", namespace=namespace, **kwargs
+    )
+
+    if not jobs:
+        return []
+
+    jobs = [
+        j
+        for j in jobs
+        if j["metadata"]["ownerReferences"][0]["uid"] == cronjob["metadata"]["uid"]
+    ]
+
+    return jobs
+
+
+def stop_jobs(cronjob_name, namespace, wait=False, timeout_seconds=60, **kwargs):
+    suspend_cronjob(cronjob_name, namespace, **kwargs)
+
+    jobs = get_jobs(cronjob_name, namespace, **kwargs)
+    for job in jobs:
+        __salt__["metalk8s_kubernetes.delete_object"](
+            kind="Job",
+            apiVersion="batch/v1",
+            name=job["metadata"]["name"],
+            namespace=namespace,
+            **kwargs,
+        )
```

Comment:
So it deletes all jobs no matter if they are running or not :thinking: it's not really a "stop job" then but ... I think it's fine ok